sigstore/sign.py

```diff
@@ -82,74 +89,106 @@ class Signer:
     The primary API for signing operations.
     """
 
-    def __init__(self, *, fulcio: FulcioClient, rekor: RekorClient):
+    def __init__(self, identity_token: str, cache: bool = True) -> None:
         """
         Create a new `Signer`.
 
-        `fulcio` is a `FulcioClient` capable of connecting to a Fulcio instance
-        and returning signing certificates.
+        `identity_token` is the identity token used to request a signing certificate
+        from Fulcio.
 
-        `rekor` is a `RekorClient` capable of connecting to a Rekor instance
-        and creating transparency log entries.
+        `cache` determines whether the signing certificate and ephemeral private key
+        should be reused (until the certificate expires) to sign different artifacts.
+        Default is `True`.
         """
-        self._fulcio = fulcio
-        self._rekor = rekor
+        self._identity_token: str = identity_token
+        self.cache: bool = cache
+        self._signing_certificate: Optional[FulcioCertificateSigningResponse] = None
+        self._private_key: Optional[ec.EllipticCurvePrivateKey] = None
+
+    @property
+    def private_key(self) -> ec.EllipticCurvePrivateKey:
+        """Get or generate a signing key."""
+        if not self.cache or self._private_key is None:
+            logger.debug("Generating ephemeral keys...")
+            self._private_key = ec.generate_private_key(ec.SECP384R1())
+
+        return self._private_key
+
+    @property
+    def signing_cert(
+        self,
+    ) -> Callable[
+        [str, ec.EllipticCurvePrivateKey, FulcioClient],
+        FulcioCertificateSigningResponse,
+    ]:
```

Comment:
I had to re-read this a few times to figure out what was happening -- it's a `@property` that returns a nested function, so subsequent callsites look like instance methods when they really aren't 🙂 

That's a little confusing IMO: this should probably either be a normal instance method (no `@property`) or a `@property` that returns `FulcioCertificateSigningResponse | None`, with `None` indicating the exceptional case (`cache and expired`). 

Thoughts?

Reply:
+1, a simple method should be enough here